discovery/gossiper.go

```diff
@@ -2029,22 +2029,37 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []n
 				remotePub, _ = chanInfo.NodeKey1()
 			}
 
-			sPeer, err := d.cfg.FindPeer(remotePub)
-			if err != nil {
-				log.Errorf("unable to send channel update -- "+
-					"could not find peer %x: %v",
-					remotePub, err)
-			} else {
-				// Send ChannelUpdate directly to remotePeer.
-				// TODO(halseth): make reliable send?
-				err = sPeer.SendMessage(false, msg)
-				if err != nil {
-					log.Errorf("unable to send channel "+
+			// Now, we'll attempt to send the channel update message
+			// reliably to the remote peer in the background, so
+			// that we don't block if the peer happens to be offline
+			// at the moment.
+			d.wg.Add(1)
+			go func() {
+				defer d.wg.Done()
+
+				var peer lnpeer.Peer
+				peerChan := make(chan lnpeer.Peer, 1)
+
+				for {
+					d.cfg.NotifyWhenOnline(remotePub, peerChan)
```

Comment:
The `AnnSig` message is added to the DB in `sendAnnSigReliably`, to make sure the resend procedure survives restarts. Should we repurpose the method to also be used for the `ChannlUpdate` message?

Reply:
Do we need to store the chan update message tho? The ann sig is unique as we don't have something that is able to store a _partially_ complete chan ann in the db. However here, we already have a complete chan update.

Reply:
We need to "remember" somehow that we did not successfully send the message. Just reusing the existing logic would entail the least among of extra work to achieve this I believe.